test/libsolidity/semanticTests/memoryManagement/assembly_access.sol

```diff
@@ -0,0 +1,14 @@
+contract C {
```

Comment:
I'd call the test something like `assembly_access_to_uninitialized_dynamic_array` to make the purpose clearer.